evalml/tests/utils_tests/test_gen_utils.py

```diff
@@ -271,6 +271,13 @@ def test_pad_with_nans(data, num_to_pad, expected):
     _check_equality(padded, expected)
 
 
+def test_pad_with_nans_with_series_name():
```

Comment:
Thanks for adding this test. I think this gives us coverage in the case where gap/max_delay != 0